src/Payone/RequestParameter/Builder/ShippingInformationRequestParameterBuilder.php

```diff
@@ -42,6 +55,20 @@ public function getRequestParameter(AbstractRequestParameterStruct $arguments):
                 'shipping_city' => $shippingAddress->getCity(),
                 'shipping_country' => $shippingAddress->getCountry()?->getIso(),
             ]);
+
+            if (\in_array($shippingAddress->getCountry()?->getIso(), self::COUNTRIES_FOR_WHICH_A_STATE_MUST_BE_SPECIFIED, true)) {
+                $countryStateCode = $shippingAddress->getCountryState()?->getShortCode();
+                if ($countryStateCode === null) {
+                    throw new \RuntimeException('missing state in shipping address');
+                }
+
+                $parts = explode('-', $countryStateCode);
+                if (\count($parts) !== 2) {
+                    throw new \RuntimeException('invalid country state code');
```

Comment:
i think it would be better to not throw an exception. Just return null, if the parts can not exploded, also if no state is set. 

so this may not break the checkout and the payment will just fail its regular way. (or may be successfully completed, in any way)

Reply:
At first I also thought about simply not transferring the value, but then the payment would fail with PAYONE. And in the other RequestParameterBuilders, an exception is also thrown if required data is missing. That's why I decided to use the exception.